### PR TITLE
Remove parent oid from find_children_oids result

### PR DESCRIPTION
**Pull request by svenklemm**
Since the parent table of a hypertable cannot contain
any rows including it in the plan is not necessary. PostgreSQL might
also not have good statistics for the main table as autoanalyze is
triggered by inserts which never happen to the main table, leading to
worse plans.

This patch adds a new guc enable_constraint_exclusion to disable
planner constraint exclusion. Constraint exclusion will not work with
cursors and WHERE CURRENT OF so the new guc will allow you to disable
that feature.

This patch also fixes constraint exclusion for JOINs and sort
optimization for queries with chunks_in.

### CODE REVIEW

File: test/expected/agg_bookends_optimized-11.out [hunk 1]
```diff
@@ -23,14 +23,12 @@ INSERT INTO "btest" VALUES('2017-01-20T09:00:21', '2017-01-20T09:00:56', 2, 30.2
...
-                QUERY PLAN                
-------------------------------------------
- Sort
-   Sort Key: btest."time"
-   ->  Append
-         ->  Seq Scan on btest
-         ->  Seq Scan on _hyper_1_1_chunk
-(5 rows)
+                                     QUERY PLAN                                      
+-------------------------------------------------------------------------------------
```
[the remark this answers is not in the document]
Comment (svenklemm) [reply test/expected/agg_bookends_optimized-11.out]: well that can happen cause postgres doesnt know before how many children it will end up with

File: test/expected/plan_ordered_append-11.out [hunk 2]
```diff
@@ -353,34 +343,36 @@ ORDER BY time ASC LIMIT 1;
   time_bucket('1d',time), device_id, value
 FROM ordered_append
 ORDER BY 1 LIMIT 1;
-                                                                 QUERY PLAN                                                                 
---------------------------------------------------------------------------------------------------------------------------------------------
+                                  QUERY PLAN
```
Comment (cevian) [inline test/expected/plan_ordered_append-11.out]: This change bothers me. It seem like the optimization isn't applied. Looking at `timescaledb_set_rel_pathlist` I see that it does expect the parent to be in the append list to apply the `ts_sort_transform_optimization` optimization.

File: test/expected/rowsecurity-11.out [hunk 3]
```diff
@@ -3475,34 +3445,31 @@ SET ROLE regress_rls_bob;
 PREPARE role_inval AS SELECT * FROM t1;
 -- Check plan
 EXPLAIN (COSTS OFF) EXECUTE role_inval;
-          QUERY PLAN           
--------------------------------
- Append
-   ->  Seq Scan on t1
-         Filter: ((a % 2) = 0)
-(3 rows)
+        QUERY PLAN
```
Comment (cevian) [inline test/expected/rowsecurity-11.out]: we probably do want to add some data to this table so that we can see the filter correctly applied. Right now these test no longer test anything.

File: test/expected/rowsecurity-11.out [hunk 4]
```diff
@@ -4178,75 +4142,28 @@ FETCH ABSOLUTE 1 FROM current_check_cursor;
 
 -- Still cannot UPDATE row 2 through cursor
 UPDATE current_check SET payload = payload || '_new' WHERE CURRENT OF current_check_cursor RETURNING *;
- currentid | payload | rlsuser 
------------+---------+---------
-(0 rows)
-
+ERROR:  cursor "current_check_cursor" is not a simply updatable scan of table "current_check"
```
Comment (cevian) [inline test/expected/rowsecurity-11.out]: These look bad. Why the change?

File: test/expected/sql_query_results_optimized.out [hunk 5]
```diff
@@ -137,42 +136,42 @@ SELECT * FROM hyper_timefunc ORDER BY unix_to_timestamp("time") DESC limit 2;
...
-                     ->  Index Scan using _hyper_4_10_chunk_time_plain_date on _hyper_4_10_chunk
-                     ->  Index Scan using _hyper_4_11_chunk_time_plain_date on _hyper_4_11_chunk
-                     ->  Index Scan using _hyper_4_12_chunk_time_plain_date on _hyper_4_12_chunk
-                     ->  Index Scan using _hyper_4_13_chunk_time_plain_date on _hyper_4_13_chunk
-                     ->  Index Scan using _hyper_4_14_chunk_time_plain_date on _hyper_4_14_chunk
-                     ->  Index Scan using _hyper_4_15_chunk_time_plain_date on _hyper_4_15_chunk
-                     ->  Index Scan using _hyper_4_16_chunk_time_plain_date on _hyper_4_16_chunk
-                     ->  Index Scan using _hyper_4_17_chunk_time_plain_date on _hyper_4_17_chunk
-                     ->  Index Scan using _hyper_4_18_chunk_time_plain_date on _hyper_4_18_chunk
+   ->  Sort
```
Comment (cevian) [inline test/expected/sql_query_results_optimized.out]: This change may also be due to the lack of sort_transform_optimization.

File: test/expected/sql_query_results_optimized.out [hunk 6]
```diff
@@ -198,19 +197,22 @@ WHERE time < to_timestamp(900)
...
-         Group Key: (date_trunc('minute'::text, hyper_1."time"))
-         ->  Custom Scan (ConstraintAwareAppend)
-               Hypertable: hyper_1
-               Chunks left after exclusion: 1
-               ->  Merge Append
-                     Sort Key: (date_trunc('minute'::text, _hyper_1_1_chunk."time")) DESC
-                     ->  Index Scan using _hyper_1_1_chunk_time_plain on _hyper_1_1_chunk
-                           Index Cond: ("time" < 'Wed Dec 31 16:15:00 1969 PST'::timestamp with time zone)
-(10 rows)
+   ->  Sort
```
Comment (cevian) [inline test/expected/sql_query_results_optimized.out]: lack of sort_transoform_optimization

File: test/expected/sql_query_results_optimized.out [hunk 7]
```diff
@@ -137,42 +136,42 @@ SELECT * FROM hyper_timefunc ORDER BY unix_to_timestamp("time") DESC limit 2;
...
-               ->  Merge Append
-                     Sort Key: (date_trunc('minute'::text, hyper_1."time")) DESC
-                     ->  Index Scan using time_plain on hyper_1
-                     ->  Index Scan using _hyper_1_1_chunk_time_plain on _hyper_1_1_chunk
+         Group Key: (date_trunc('minute'::text, _hyper_1_1_chunk."time"))
+         ->  Sort
+               Sort Key: (date_trunc('minute'::text, _hyper_1_1_chunk."time")) DESC
+               ->  Result
+                     ->  Append
+                           ->  Seq Scan on _hyper_1_1_chunk
```
Comment (cevian) [inline test/expected/sql_query_results_optimized.out]: looks like lack of sort_transform opt too

File: test/expected/sql_query_results_optimized.out [hunk 8]
```diff
@@ -323,12 +321,12 @@ FROM hyper_1 GROUP BY t ORDER BY t DESC limit 2;
...
  Limit
    ->  GroupAggregate
-         Group Key: (time_bucket('@ 1 min'::interval, hyper_1."time"))
-         ->  Result
-               ->  Merge Append
-                     Sort Key: (time_bucket('@ 1 min'::interval, hyper_1."time")) DESC
-                     ->  Index Scan using time_plain on hyper_1
-                     ->  Index Scan using _hyper_1_1_chunk_time_plain on _hyper_1_1_chunk
+         Group Key: (time_bucket('@ 1 min'::interval, _hyper_1_1_chunk."time"))
+         ->  Sort
```
Comment (cevian) [inline test/expected/sql_query_results_optimized.out]: lacks sort_transform_optimization

File: test/expected/sql_query_results_optimized.out [hunk 9]
```diff
@@ -123,10 +123,9 @@ EXPLAIN (costs off) SELECT * FROM hyper_timefunc ORDER BY unix_to_timestamp("tim
 -----------------------------------------------------------------------------------------
```
Comment (cevian) [inline test/expected/sql_query_results_optimized.out]: Note to self: This whole file needs to be re-checked after sort transform optimization is fixed.
